sycl/include/sycl/vector.hpp

```diff
@@ -1194,37 +1180,25 @@ class SwizzleOp {
     return NewRelOp<RhsOperation, LogicalOr, Indexes...>(NULL, *this, Rhs);
   }
 
-  // Begin hi/lo, even/odd, xyzw, and rgba swizzles.
 private:
-  // Indexer used in the swizzles.def.
-  // Currently it is defined as a template struct. Replacing it with a constexpr
-  // function would activate a bug in MSVC that is fixed only in v19.20.
-  // Until then MSVC does not recognize such constexpr functions as const and
-  // thus does not let using them in template parameters inside swizzle.def.
-  template <int Index> struct Indexer {
-    static constexpr int IDXs[sizeof...(Indexes)] = {Indexes...};
-    static constexpr int value = IDXs[Index >= getNumElements() ? 0 : Index];
-  };
+  static constexpr auto get_vec_idx(int idx) {
```

Comment:
AFAI understood this function returns different value than the old one. Old one returns Indexes[0] if idx > size of Indexes. New function returns -1, is it expected?

Reply:
Yes, although both `vec` and `swizzle` are missing SFINAE/constraints on `swizzle()` methods that should be catching that. I'll fix that issue separately.

Named swizzles have proper constraints so wrong indices aren't possible there.